x-pack/plugins/snapshot_restore/public/app/sections/home/snapshot_list/snapshot_details/snapshot_details.tsx

```diff
@@ -0,0 +1,361 @@
+                defaultMessage="Failures ({failuresCount})"
+                values={{ failuresCount: failures.length }}
+              />
+            </EuiDescriptionListTitle>
+
+            <EuiDescriptionListDescription
+              className="eui-textBreakWord"
+              data-test-subj="srSnapshotDetailFailuresDescription"
+            >
+              {failuresList}
+            </EuiDescriptionListDescription>
+          </EuiFlexItem>
+        </EuiFlexGroup>
+
+        <EuiFlexGroup>
+          <EuiFlexItem data-test-subj="srSnapshotDetailsStartTimeItem">
+            <EuiDescriptionListTitle>
+              <FormattedMessage
+                id="xpack.snapshotRestore.snapshotDetails.itemStartTimeLabel"
+                data-test-subj="srSnapshotDetailsStartTimeTitle"
+                defaultMessage="Start time"
+              />
+            </EuiDescriptionListTitle>
+
+            <EuiDescriptionListDescription
+              className="eui-textBreakWord"
+              data-test-subj="srSnapshotDetailStartTimeDescription"
+            >
+              {formatDate(startTimeInMillis)}
+            </EuiDescriptionListDescription>
+          </EuiFlexItem>
+
+          <EuiFlexItem data-test-subj="srSnapshotDetailsEndTimeItem">
+            <EuiDescriptionListTitle>
+              <FormattedMessage
+                id="xpack.snapshotRestore.snapshotDetails.itemEndTimeLabel"
+                data-test-subj="srSnapshotDetailsEndTimeTitle"
+                defaultMessage="End time"
+              />
+            </EuiDescriptionListTitle>
+
+            <EuiDescriptionListDescription
+              className="eui-textBreakWord"
+              data-test-subj="srSnapshotDetailEndTimeDescription"
+            >
+              {formatDate(endTimeInMillis)}
+            </EuiDescriptionListDescription>
+          </EuiFlexItem>
+        </EuiFlexGroup>
+
+        <EuiFlexGroup>
+          <EuiFlexItem data-test-subj="srSnapshotDetailsDurationItem">
+            <EuiDescriptionListTitle>
+              <FormattedMessage
+                id="xpack.snapshotRestore.snapshotDetails.itemDurationLabel"
+                data-test-subj="srSnapshotDetailsDurationTitle"
+                defaultMessage="Duration"
+              />
+            </EuiDescriptionListTitle>
+
+            <EuiDescriptionListDescription
+              className="eui-textBreakWord"
+              data-test-subj="srSnapshotDetailDurationDescription"
+            >
+              <FormattedMessage
+                id="xpack.snapshotRestore.snapshotDetails.itemDurationValueLabel"
+                data-test-subj="srSnapshotDetailsDurationValue"
+                defaultMessage="{seconds} seconds"
+                values={{ seconds: Math.round(durationInMillis / 1000) }}
+              />
+            </EuiDescriptionListDescription>
+          </EuiFlexItem>
+        </EuiFlexGroup>
+      </EuiDescriptionList>
+    );
+  } else if (error) {
```

Comment:
I think the error condition should come before the details condition. when I open a snapshot's details and manually change the snapshot ID in the URL to a snapshot ID that does not exist, this UI doesn't surface.

changing L65 to `if (!error && snapshotDetails)` also works